packages/typescript/internal/ts_project.bzl

```diff
@@ -323,6 +323,9 @@ def _is_ts_src(src, allow_js):
         return True
     return allow_js and (src.endswith(".js") or src.endswith(".jsx"))
 
+def _is_json_src(src):
+    return src.endswith(".json")
```

Comment:
I am including JSON srcs unconditionally. This _may_ be fine as `tsc` will error if JSON files are imported without `resolveJsonModule`.

On the other hand, they could be included conditionally on `resolveJsonModule` being set to `True` either in an attribute on the rule or nested in the tsconfig.compilerOptions map. This would be similar to how `allowJs` drives the inclusion of JS files.

Reply:
I think unconditional inclusion may cause a problem if people are globing `**/*` and this sucks up incidental JSON files. I'm not sure how `tsc` behaves if JSON files are included in its compilation without being imported from any TS files. Whatever the case, it may become more or less lenient in future, so maybe we should play safe here.

Any thoughts?

Reply:
We definitely need to handle those corner cases correctly. If you're not sure how `tsc` behaves then do that experiment, or add that scenario under packages/typescript/test/ts_project and let's verify?

Reply:
ok, should be good now. I went down the route of:

1. adding a new attribute to the `ts_project` macro & rule, `resolve_json_module`
2. only including json srcs if `resolve_json_module` is true, as happens for `allow_js`
3. auto-including json srcs if `srcs` is None and `resolve_json_module` is true, as happens for `allow_js`

I did notice some inconsistency as to how some of the attributes are handled versus the equivalent inside `ts_config.compilerOptions`, but that is more general than this so I'd like to probe that in a separate PR.